api.js

```diff
@@ -26,6 +26,14 @@ module.exports = async (filePaths, {find, replacement, ignoreCase} = {}) => {
 		throw new Error('The `replacement` option is required');
 	}
 
+	// Replace replacement string with string unescaped (only one backslash) if it is escaped
+	replacement = replacement
+		.replace(/\\n/g, '\n')
```

Comment:
Small suggestion to simplify the replacement:

```js
replacement.replace(/\\([\n\r\t\b\f])/, ‘\$2’);
```

I haven’t tested it because I’m writing this one my phone, but the idea is to use the capture group.

Can you also explain why are should handle \b and \f?

Reply:
Another thing to have into consideration is that we should make the regexp global, so that multiple new lines are correctly replaced. If you make a test where the replacement is ‘\n\n’, you will notice why we need it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp - look for the global flag/modifier

Reply:
After some research I don't think \f should be handled, I added it because I thought about escaping every single one that exists, but in this case I don't think it's relevant. \b could've been relevant but in this case since it's an input I don't believe it can be useful, but I could be wrong. 

As for the regular expression, I've been trying to resolve the problem with it but it still doesn't write the expected output. I believe it's because this way the replace returns a string which  with two ASCII codes (92 and 110): 
![1](https://user-images.githubusercontent.com/28600773/61923165-e7a86100-af5a-11e9-97fc-554be185f046.PNG),
two chars, which are '\' + 'n'. And "\n"'s ASCII code is 10: 
![2](https://user-images.githubusercontent.com/28600773/61923186-f1ca5f80-af5a-11e9-935b-e4e88a80d026.PNG)
and it's only one char. Therefore I believe the best way to surpass the problem is by replacing the whole string like I did before with the corresponding char since I don't know a way to transform the output string into a char without resorting to individual clauses for each case.

I added a test too with multiple \n \t and \r to test if the regexp is global.

Reply:
👍